### ISSUE TITLE
Changes get reverted in the html mode after clicking a link

### ISSUE THREAD

**Issue by nhithanki** [craftcms/redactor#419]
### Description
After clicking any link and turn on html mode to edit, whenever finishing editing and click one more time, all changes will be removed.

### Steps to reproduce

1.Click a link.
2. Turn on html mode.
3. Type any random character.
4. Click again.

### Additional info

- Craft version: Craft Solo 3.7.46
- PHP version: 7.4.22
- Database driver & version: MySQL 8.0.28
- Plugins & versions: 2.10.8

**Comment by brandonkelly** [comment]
I’m not sure I’m following. Can you give an example of a link that you are clicking on? In general links are supposed to navigate you away from the page, so I’m not sure how you are editing the raw HTML after doing so. Do you mean clicking on a link within the Redactor field content?

**Comment by nhithanki** [comment]
Yes. Clicking on a link in redator field then open html mode and editing
something.
After that, just randomly clicking one more time.
All the change inside the html mode will be reverted.

**Comment by nhithanki** [comment]
https://user-images.githubusercontent.com/40749692/176577435-90b17911-259e-4fbe-adc5-aaace921ed50.mp4



> I’m not sure I’m following. Can you give an example of a link that you are clicking on? In general links are supposed to navigate you away from the page, so I’m not sure how you are editing the raw HTML after doing so. Do you mean clicking on a link within the Redactor field content?

**Comment by brandonkelly** [comment]
Thanks for clarifying! Was able to reproduce, and have just released Redactor 2.10.10 (Craft 3) and 3.0.2 (Craft 4) with a fix.
